src/firmware/main/world/firmware_ball.h

```diff
@@ -0,0 +1,65 @@
+#pragma once
+
+/**
+ * This struct represents the ball from the perspective of firmware
+ */
+struct FirmwareBall;
+typedef struct FirmwareBall FirmwareBall;
+
+/**
+ * Create a ball with methods for accessing it's state
+ *
+ * @param get_ball_position_x A function that can be called to get the x-position of the
```

Comment:
Nit:
What is the reference point for the X-Y plane?

Reply:
Hmm, so it's general convention that `(0,0)` is the center of the field. Any particular reason we should specify it here, given that we don't elsewhere?

Reply:
I think it might be reasonable to specify if it's the "global" world/field reference frame vs the robot's local reference frame. Generally I don't think we need to specify the origin (people should read the doc about coordinate conventions), but because this is firmware people may assume it's the robot's frame of reference to we should be clear.

Reply:
added notes here and in `firmware_robot.h`